src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs

```diff
@@ -94,6 +94,34 @@ protected bool Idle(object message)
             }
         }
 
+        protected void RequestQueryPermit()
+        {
+            Log.Debug("Requesting query permit");
+            QueryPermitter.Tell(RequestQueryStart.Instance);
+            Become(WaitingForQueryPermit);
+        }
+
+        protected bool WaitingForQueryPermit(object message)
```

Comment:
All queries now run with an execution flow that looks similar to this:

```mermaid
stateDiagram-v2
    [*] --> Init: Request (from Akka.Streams)
    Init --> RequestQueryPermit: RequestQueryStart
    RequestQueryPermit --> Replay: QueryStartGranted
    Replay --> Replay: ReplayedMessage
    Replay --> Idle: RecoverySuccess (return token to permitter)
    Idle --> RequestQueryPermit: Continue
```

At each stage in the query, we need permission from the QueryPermitter to continue - this is designed to help rate-limit the number of queries that can hit the database at any given time.